src/generator/AutoRest.Ruby.Azure/CodeGeneratorRba.cs

```diff
@@ -66,14 +66,14 @@ public CodeGeneratorRba()
                     continue;
                 }
 
-                if( codeModel.pageModels.Any( each => each.Name.EqualsIgnoreCase(model.Name ) ) )
+                if ( codeModel.pageModels.Any( each => each.Name.EqualsIgnoreCase(model.Name ) ) )
                 {
                     // Skip, handled in the .pageModels section below.
                     continue;
                 }
 
                 var modelTemplate = new AzureModelTemplate { Model = model };
-                if(!CompositeTypeRba.nameRegEx.IsMatch(model.Name.ToString()) || !CompositeTypeRba.IsResourceModelsMatchStandardDefinition(model))
+                if (!CompositeTypeRba.resourceOrSubResourceRegEx.IsMatch(model.Name.ToString()) || !CompositeTypeRba.IsResourceModelMatchingStandardDefinition(model))
```

Comment:
This was missed in my previous change. You do not need to convert it to string. Changed it.